src/core/search/sort_indices.cc

```diff
@@ -66,18 +71,23 @@ template <typename T> PMR_NS::memory_resource* SimpleValueSortIndex<T>::GetMemRe
 template struct SimpleValueSortIndex<double>;
 template struct SimpleValueSortIndex<PMR_NS::string>;
 
-double NumericSortIndex::Get(DocId id, DocumentAccessor* doc, std::string_view field) {
+std::optional<double> NumericSortIndex::Get(DocId id, DocumentAccessor* doc,
+                                            std::string_view field) {
   auto str = doc->GetStrings(field);
   if (str.empty())
-    return 0;
+    return 0.0;
 
-  double v;
-  if (!absl::SimpleAtod(str.front(), &v))
-    return 0;
-  return v;
+  auto value_as_double = ParseNumericField(str.front());
+  if (!value_as_double) {
+    LOG(WARNING) << "Failed to parse numeric value from field: " << field
```

Comment:
why it's a warning? can it happen as a result of user input? is it a bug or a valid scenario?

Reply:
Yes, it can happen as a result of user input

Reply:
I think it would be better to remove this `LOG`